src/shogun/clustering/KMeansBase.cpp

```diff
@@ -362,8 +314,10 @@ void CKMeansBase::init()
 	SG_ADD(&max_iter, "max_iter", "Maximum number of iterations", ParameterProperties::HYPER);
 	SG_ADD(&k, "k", "k, the number of clusters", ParameterProperties::HYPER);
 	SG_ADD(&dimensions, "dimensions", "Dimensions of data");
+	SG_ADD(&fixed_centers, "fixed_centers", "Use fixed centers");
 	SG_ADD(&R, "radiuses", "Cluster radiuses");
 	SG_ADD(&use_kmeanspp, "kmeanspp", "Whether use kmeans++", ParameterProperties::HYPER);
+	SG_ADD(&mus, "mus", "Cluster centers")
```

Comment:
@karlnapf I am a bit confused by this variable here. From what I could understand from the code, it should contain the actual centroids. However, I noticed that there is also a method called `get_cluster_centers`, which computes the centroids from the `distance`. Why is it that? Aren't they the same thing?

Reply:
the cluster centers are stored in the lhs of some feature matrix. I think that is bad. Should be refactored at some point.
I think you can sidestep this via doing an explicit observe call on the result of that method. Once this is refactored and the cluster centers are just stored in a variable, I think it is better to observe them with a simple put, or?